### DISCUSSION TITLE
How to add types for custom variants on MantineProvider?

### DISCUSSION THREAD

**Discussion by jason-dark** [mantinedev/mantine#4511] [Q&A] [answered]
Wondering if anyone can help me with this question.

When adding custom colours in a Mantine Typescript project we can do [this](https://mantine.dev/theming/colors/#add-custom-colors-types) to get typings working correctly. 

My project has some custom buttons defined using the variants method described [here](https://mantine.dev/theming/mantine-provider/#variants-on-mantineprovider).

Is there any way to define the types for custom button variants too, like we do for colours?

**Comment by rtivital** [comment, marked as answer]
No, augmentation of other types is not supported
